python/xir/parser.py

```diff
@@ -347,5 +346,5 @@ def is_wire(arg):
 
 
 def is_param(arg):
-    """Returns whether the passed argument is a list of params."""
+    """Returns whether the passed argument is a list or dictionary of params."""
```

Comment:
Are you sure the implementation of this should still check for `"params"` and not
```python
return isinstance(arg, tuple) and arg[0] in ("params_dict", "params_list")
```

Reply:
All params (no matter if they're dicts or lists) will be coupled with the "params" string identifier. I think this is fine since it's easy to check whether its a list or a dict. It could also be changed to have two different identifiers "params_list" and "params_dict", which would work as well.